### ISSUE TITLE
"AssertError:Allreduce: boundary error" when using hyperopt

### ISSUE THREAD

**Issue by edi-bice** [dmlc/xgboost#2410]
training with parameters from parameter grid works fine

params = {'colsample_bytree': 0.85, 'eval_metric': 'auc', 'max_delta_step': 0.2, 'min_child_weight': 3, 'n_estimators': 750, 'subsample': 0.85, 'objective': 'binary:logistic', 'max_depth': 7, 'gamma': 0.5, 'booster': 'gbtree'}
 bst = xgb.train(params, dtrain, num_round, watchlist, maximize=True, callbacks=[xgb.callback.print_evaluation(show_stdv=True), xgb.callback.early_stop(50), xgb.callback.reset_learning_rate(lrs)])

training with parameters from hyperopt space doesn't

params = {'eval_metric': 'auc', 'min_child_weight': 5, 'subsample': 0.8, 'seed': 123456, 'objective': 'binary:logistic', 'max_depth': 8, 'gamma': 0.8, 'booster': 'gbtree'}
bst = xgb.train(params, dtrain, num_round, watchlist, maximize=True, early_stopping_rounds=50)

[20:35:33] 2278501x212 matrix with 263224281 entries loaded from hdfs://xxx/yyy/data.libsvm
[20:36:50] src/tree/updater_prune.cc:74: tree pruning end, 1 roots, 128 extra nodes, 0 pruned nodes, max_depth=7
AssertError:Allreduce: boundary error

nohup ~/xgboost/dmlc-core/tracker/dmlc-submit --cluster=yarn --num-workers=27 --worker-cores=6 --worker-memory 20g /opt/rh/python27/root/usr/bin/python ~/precogmodeling/fraudhy.py 2>&1 >>fraudhy.log_nohup &

RedHat 6.9, gcc (GCC) 6.3.1 
xgboost commit fd19b7a188a8b048f372827650b1111d4f8b1fa5
